batch/batches/MoveCategoryEntries/KAsyncMoveCategoryEntries.class.php

```diff
@@ -187,79 +139,75 @@ private function moveEntries(KalturaBatchJob $job, KalturaMoveCategoryEntriesJob
-		while(count($categoryEntriesList->objects))
-		{
-			$entryIds = array();
-			$ancestor = $data->destCategoryId;
-			if(array_key_exists($ancestor, $this->ancestors))
-				$ancestor = $this->ancestors[$ancestor];
-			
-			$addedCategoryEntriesResults = $this->addCategoryEntries($categoryEntriesList, $ancestor, $entryIds);
-			
-			$categoryDeleted = false;
-			if(	is_array($addedCategoryEntriesResults[0]) && isset($addedCategoryEntriesResults[0]['code']) && ($addedCategoryEntriesResults[0]['code'] == self::CATEGORY_NOT_FOUND))
-				$categoryDeleted = true;
-			
-			
-			if($categoryDeleted) {
-				$ancestor = $this->getDeepestLiveAncestor($data->destCategoryId, $data->destCategoryFullIds, true);
-				// In case the category isn't found since it was just deleted, recall with the matching ancestor
-				if($ancestor) {
-					$data->destCategoryId = $ancestor;
-					continue;
+		while(count($categoryEntriesList->objects)) {
+				$entryIds = array();
+				$categoryIds = array();
+				$ancestor = $data->destCategoryId;
+				if (array_key_exists($ancestor, $this->ancestors)) {
+					$ancestor = $this->ancestors[$ancestor];
 				}
-			}
-			
-			KBatchBase::$kClient->startMultiRequest();
-			foreach($addedCategoryEntriesResults as $index => $addedCategoryEntryResult)
-			{
-				$code = null;
-				if(	is_array($addedCategoryEntryResult) && isset($addedCategoryEntryResult['code']))
-					$code = $addedCategoryEntryResult['code'];
-						
-				if(!is_null($code) && !in_array($code, array(self::CATEGORY_ENTRY_ALREADY_EXISTS, self::INVALID_ENTRY_ID, self::CATEGORY_NOT_FOUND)))
-					continue;
-					
-				if($data->copyOnly)
-					continue;
-					
-				KBatchBase::$kClient->categoryEntry->delete($entryIds[$index], $srcCategoryId);
-			}
-			$deletedCategoryEntriesResults = KBatchBase::$kClient->doMultiRequest();
-			if(is_null($deletedCategoryEntriesResults))
-				$deletedCategoryEntriesResults = array();
-			
-			foreach($deletedCategoryEntriesResults as $index => $deletedCategoryEntryResult)
-			{
-				if(is_array($deletedCategoryEntryResult) && isset($deletedCategoryEntryResult['code']))
-				{
-					KalturaLog::err('error: ' . $deletedCategoryEntryResult['code']);
-					unset($deletedCategoryEntriesResults[$index]);
+				$addedCategoryEntriesResults = $this->addCategoryEntries($categoryEntriesList, $ancestor, $entryIds, $categoryIds);
+
+				$categoryDeleted = false;
+				if (is_array($addedCategoryEntriesResults[0]) && isset($addedCategoryEntriesResults[0]['code']) && ($addedCategoryEntriesResults[0]['code'] == self::CATEGORY_NOT_FOUND)) {
+					$categoryDeleted = true;
+				}
+
+				if ($categoryDeleted) {
+					$ancestor = $this->getDeepestLiveAncestor($data->destCategoryId, $data->destCategoryFullIds, true);
+					// In case the category isn't found since it was just deleted, recall with the matching ancestor
+					if ($ancestor) {
+						$data->destCategoryId = $ancestor;
```

Comment:
The job is not supposed to change the input data, only output data if there is any.